### PR TITLE
Update IContentFinder code samples

### PR DESCRIPTION
**Pull request by erikjanwestendorp**
Fixes: https://github.com/umbraco/UmbracoDocs/issues/4111

return `Task<bool>` instead of `bool`.

### CODE REVIEW

**Comment by eshanrnh** [comment]
Thanks @erikjanwestendorp for the PR 💪

We will review this soon 😃

**Comment by bergmania** [comment]
Hi @erikjanwestendorp. Thanks for the PR.

I would suggest using this pattern `Task.FromResult(false)` instead of  this `new Task<bool>(() => false)`.

What do you think?

**Comment by erikjanwestendorp** [comment]
Hi @bergmania Thanks for your review and suggestion 😄. 

Just updated the PR.

File: Reference/Routing/Request-Pipeline/IContentFinder.md [hunk 1]
```diff
@@ -180,33 +179,37 @@ namespace RoutingDocs.ContentFinders
...
-                return false;
+                return Task.FromResult(false);
             }
 
             // Assuming the 404 page is in the root of the language site with alias fourOhFourPageAlias
-            var notFoundNode = siteRoot.Children.FirstOrDefault(f => f.ContentType.Alias == "fourOhFourPageAlias");
+            var notFoundNode = siteRoot.Children?.FirstOrDefault(f => f.ContentType.Alias == "fourOhFourPageAlias");
 
-            if (notFoundNode is not null)
+            if (notFoundNode != null)
```
Comment (bergmania) [inline Reference/Routing/Request-Pipeline/IContentFinder.md:206]: Not sure about this.. `!=` is less effective then `is not` due to possible operation overloads..

Comment (bergmania) [inline Reference/Routing/Request-Pipeline/IContentFinder.md:206]:
```suggestion
            if (notFoundNode is not null)
```

**Comment by bergmania** [comment]
Thanks.. Just found a small thing more.. Not a big deal, more a general thing :)

**Comment by erikjanwestendorp** [comment]
@bergmania Thanks again just committed your suggestion.

**Review by bergmania — APPROVED** [review]

**Comment by eshanrnh** [comment]
Merging it in 😀

**Pull request merged**